### PR TITLE
WIP 7.1 signatures

### PR DESCRIPTION
**Pull request by greg0ire**
I am targeting this branch, because this is a huge BC-break.

## Changelog

```markdown
### Changed
- most method signatures now have php 7.1 type hints
```

## To do

- [ ] Change signatures in interfaces that are in the admin bundle :/
- [ ] Change more signatures
- [ ] Add an upgrade note

### CODE REVIEW

**Comment by core23** [comment]
Ping @greg0ire

File: src/Model/ModelManager.php [hunk 1]
```diff
@@ -363,11 +357,14 @@ public function getIdentifierValues($entity)
...
     }
 
-    public function getIdentifierFieldNames($class)
+    public function getIdentifierFieldNames($class): array
     {
         return $this->getMetadata($class)->getIdentifierFieldNames();
     }
 
+    /**
+     * @return string|void
```
Comment (core23) [inline src/Model/ModelManager.php]: string or **void**? This looks wrong

File: src/Model/ModelManager.php [hunk 2]
```diff
@@ -399,6 +396,8 @@ public function getNormalizedIdentifier($entity)
      *
      * The ORM implementation does nothing special but you still should use
      * this method when using the id in a URL to allow for future improvements.
+     *
+     * @return string|void
```
Comment (core23) [inline src/Model/ModelManager.php]: Same here

**Comment by greg0ire** [comment]
> Ping @greg0ire

no need to work on this until the first element of the todo list is resolved I'm afraid :sweat_smile:

**Comment by core23** [comment]
I was just looking at the creation date of this PR:  Dec. 2017 oO

**Comment by greg0ire** [comment]
Yeah me too, I was a bit suprised, so I started working on it, and then I saw that there were many cases where I just could not add type declarations because we rely on interfaces in the admin bundle that don't have type declarations yet.

**Pull request closed without merging**